compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Markdown.hs

```diff
@@ -241,3 +85,55 @@ escapeMd = T.pack . concatMap escapeChar . T.unpack
         | otherwise = [c]
 
     shouldEscape = (`elem` ("[]*_~`<>\\&" :: String))
+
+-- | Render fields as a pipe-table, like this:
+-- >  | Field    | Type     | Description
+-- >  | :------- | :------- | :----------
+-- >  | anA      | a        |
+-- >  | another  | a        | another a
+-- >  | andText  | Text     | and text
+renderMdFields :: RenderEnv -> [(RenderText, RenderText, RenderText)] -> [T.Text]
+renderMdFields _ []  = ["(no fields)"]
+renderMdFields env fields = header <> fieldRows
+  where
+    textFields =
+        [ ( renderMdText env name
+          , renderMdText env ty
+          , renderMdText env doc
+          )
+        | (name, ty, doc) <- fields
+        ]
+
+    fLen = maximum $ 5 :
+        [ max (T.length name) (T.length ty)
+        | (name, ty, _) <- textFields ]
+
+    header =
+        [ T.concat
+            [ "| "
+            , adjust fLen "Field"
```

Comment:
Oh I guess it might be because `length "Field" = 5`? In that case, maybe just use `maximum $ length "Field" : length "Type" : …`.

Reply:
Yes, that's right!